src/Wizard.js

```diff
@@ -60,19 +67,37 @@ const Wizard = ({ children, onComplete, debug }: Props) => {
     } else {
       nextAction();
     }
+
+    if (_stateManager) {
+      const currentStep = steps[index];
+      const nextStep = steps[index + 1];
+      _stateManager.updateHistory(currentStep.name, nextStep.name);
+    }
   }
 
   function onPrevious() {
     const prev = findPreviousValid(steps, index);
     setIndex(prev);
+
+    if (_stateManager) {
+      const currentStep = steps[index];
+      const previousStep = steps[index - 1];
+      _stateManager.updateHistory(currentStep.name, previousStep.name);
+    }
   }
 
   useEffect(() => {
     // for debugging purposes only
     if (debug) {
       console.debug('steps updated', steps); // eslint-disable-line
     }
-  }, [steps]);
+
+    if (_stateManager) {
+      const activeStep = _stateManager.getActiveStep();
+      const activeIndex = steps.findIndex(step => step.name === activeStep);
+      setIndex(activeIndex > -1 ? activeIndex : 0);
+    }
+  }, [steps, index, setIndex]);
```

Comment:
`setIndex` is a `const` and will never change, right?

Reply:
Correct! I've added `react-hooks/exhaustive-deps` now so these mistakes are less likely to happen.